src/cmd/singularity/cli/build.go

```diff
@@ -56,9 +56,12 @@ var BuildCmd = &cobra.Command{
 	// TODO: Can we plz move this to another file to keep the CLI the CLI
 	Run: func(cmd *cobra.Command, args []string) {
```

Comment:
Would it be possible to refactor this a bit into the main logic, and sub-functions? I can follow the nested if-else but it's not easy across such a long space.

Reply:
I was planning on cleaning this up and making a build.go file in libexec because of the TODO in this file. Should just do that in this PR or a separate one?